### DISCUSSION TITLE
Refactoring Traits

### DISCUSSION THREAD

**Discussion by xQwexx** [GrapesJS/grapesjs#4445] [Ideas]
So I am looking into how the traits are set up and I think the Trait model itself are unnecessary hoops.

```
    type: 'text', label: '', name: '', min: '',  max: '',  unit: '',  step: 1,   value: '',
    target: '',  default: '',  placeholder: '',  changeProp: 0,
```
These are the attributes for the trait model but everything are view specific parameters, so they should be part of the view attributes. The real model I think is the attribute of the Component (but we could not handle attributes so we should use the component as a model) so based on this the trait is more or less a config for the TraitViews.

So I am thinking to removing the trait models and use the Component as Model in the TraitView to communicate the value change directly.
This not changing the way how the traits are configured now.

**Comment by artf** [comment]
Yeah we can definitely make some improvements here and I agree about what you're saying but without the model for traits I'm wondering how would you handle [trait related methods on the component](https://grapesjs.com/docs/api/component.html#gettraits). The consumer still should be able to update/add/remove traits dynamically. Loading directly the View on init will also break the headless option of the editor :/
